src/main/java/com/mitchej123/hodgepodge/SimulationDistanceHelper.java

```diff
@@ -0,0 +1,332 @@
+package com.mitchej123.hodgepodge;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.BiPredicate;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.world.ChunkCoordIntPair;
+import net.minecraft.world.NextTickListEntry;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import net.minecraftforge.common.ForgeChunkManager;
+
+import com.mitchej123.hodgepodge.config.FixesConfig;
+import com.mitchej123.hodgepodge.config.TweaksConfig;
+
+import it.unimi.dsi.fastutil.longs.Long2BooleanOpenHashMap;
+import it.unimi.dsi.fastutil.longs.Long2ByteOpenHashMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.longs.LongBooleanPair;
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+
+public class SimulationDistanceHelper {
+
+    /**
+     * Mark a chunk as no to be simulated, or reset that state.
+     */
+    public static void preventChunkSimulation(World world, long packedChunkPos, boolean prevent) {
+        if (!FixesConfig.addSimulationDistance) {
+            return;
+        }
+        ISimulationDistanceWorld mixin = (ISimulationDistanceWorld) world;
+        mixin.hodgepodge$preventChunkSimulation(packedChunkPos, prevent);
+    }
+
+    public static int getSimulationDistance() {
+        return TweaksConfig.simulationDistance;
+    }
+
+    public static void setSimulationDistance(int distance) {
+        TweaksConfig.simulationDistance = distance;
+    }
+
+    public SimulationDistanceHelper(World world) {
+        this.world = world;
+        isServer = world instanceof WorldServer;
+    }
+
+    /**
+     * The current world
+     */
+    private final World world;
+
+    /**
+     * True if we have a WorldServer
+     */
+    private final boolean isServer;
+
+    /**
+     * Chunks that should not be ticked
+     */
+    private final Long2ByteOpenHashMap noTickChunks = new Long2ByteOpenHashMap();
+
+    /**
+     * Changes to the chunks that should not be ticked. Needed since we only update the real map once a tick. Reason is
+     * that forced chunks are also cached and updated once a tick, we want both in sync.
+     */
+    private final List<LongBooleanPair> noTickChunksChanges = new ArrayList<>();
```

Comment:
Having a separate LongArrayList and BooleanArrayList that are manually kept in sync would probably be a bit more efficient

Reply:
Alright, pushed an update